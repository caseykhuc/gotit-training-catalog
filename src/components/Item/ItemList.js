import React from 'react';
import { ListGroup } from 'react-bootstrap'
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ModifyButton from './ModifyButton';

export const ItemList = ({ items, userCurrent, categoryId }) => {
  const renderItems = () => items.map(({
    id, name, description, price, userId,
  }) => (
      <ListGroup.Item className="d-flex justify-content-between align-items-center" key={id}>
        <Link to={`/categories/items/${categoryId}/${id}`} key={id}>
          {name}
        </Link>
        {userId === userCurrent && (
          <ModifyButton
            categoryId={categoryId}
            itemId={id}
            currentValue={{ name, description, price }}
          />
        )}
      </ListGroup.Item>

    ));

  return (
    <ListGroup>
      {renderItems()}
    </ListGroup>
  );
};

ItemList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    userId: PropTypes.number.isRequired,
  })).isRequired,
  categoryId: PropTypes.number.isRequired,
  userCurrent: PropTypes.number,
}

const mapStateToProps = ({ user }) => ({
  userCurrent: user.userId,
})

export default connect(mapStateToProps)(ItemList);
