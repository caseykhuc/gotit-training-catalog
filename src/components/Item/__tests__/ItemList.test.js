
import React from 'react';
import { shallow } from 'enzyme';
import { ButtonGroup } from 'react-bootstrap';
import { ItemList } from '../ItemList';

describe('component/ItemList', () => {
  let props;
  let wrapper;
  let btnGroup;

  const update = () => {
    wrapper.update();
    btnGroup = wrapper.find(ButtonGroup);
  };
  const setup = () => {
    wrapper = shallow(<ItemList {...props} />);
    update();
  };
  beforeEach(() => {
    props = {
      items: [{
        name: 'test',
        id: 0,
        user_id: 3,
      }],
    }
  });
  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render btnGroup when userId match', () => {
    props.userId = 3;
    setup();
    expect(btnGroup.length).toBeTruthy();
  });
})
