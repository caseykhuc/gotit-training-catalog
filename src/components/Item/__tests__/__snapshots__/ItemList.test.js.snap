// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/ItemList should render correctly 1`] = `ShallowWrapper {}`;
